distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso

```diff
@@ -616,14 +616,25 @@ type Instrumentor
        ADVANCED
 
        Registers callback to be executed when a node/expression evaluation
-       is over. The callback `fn` gets UUID and the computed value. Usually
+       is over. The callback `fn` gets UUID and the computed value (or value
+       of `expression` if specified). Usually
        the value is _cached_ and returned from `on_enter` callback next time
        the same expression is evaluated.
 
+       > Example
+         Specify `expression` to _"inline evaluate"_ it.
+            see_a_b uuid:Text ~result =
+              if uuid == "expected-uuid" then
+                IO.println "evalutated to "+result.to_text
+
+            Meta.meta .fn . instrument . on_return fn=see_a_b expression="a+b" . activate
```

Comment:
Does this seem right/acceptable, @4e6?

Reply:
So, the `on_return` callback now takes an
- `fn uuid ~result` - a function to execute when the node finished
- `expression` - an expression to execute and provide its value as a `result` argument of the `fn` callback

And the `result` argument needs to be lazy to only compute the expression when the `result` value is accessed.

Took me a while to figure it out. But anyway, if it works, it works. We will probably change it later anyway when it comes to the integration with IDE

Reply:
Yes, my original version was evaluating `result` eagerly and that was failing as neither `a` nor `b` was defined.